src/Tools/ExternalAccess/VisualDiagnostics/Internal/VisualDiagnosticsServiceFactory.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Data.SqlTypes;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.VisualDiagnostics.Contracts;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Debugger.Contracts.EditAndContinue;
+using Microsoft.VisualStudio.Debugger.Contracts.HotReload;
+using Roslyn.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VisualDiagnostics;
+
+/// <summary>
+/// LSP Service responsible for hooking up to the debugger's broker IHotReloadSessionNotificationService
+/// and listening to start/end debugging session delegating to IVisualDiagnosticsLanguageService workspace service
+/// for debugger process referencing Maui.Essentials.dll
+/// </summary>
+[ExportCSharpVisualBasicLspServiceFactory(typeof(OnInitializedService)), Shared]
+internal sealed class VisualDiagnosticsServiceFactory : ILspServiceFactory
+{
+    private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService;
+    private readonly Lazy<IBrokeredDebuggerServices> _brokeredDebuggerServices;
+
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    [ImportingConstructor]
+    public VisualDiagnosticsServiceFactory(
+        LspWorkspaceRegistrationService lspWorkspaceRegistrationService,
+        Lazy<IBrokeredDebuggerServices> brokeredDebuggerServices)
```

Comment:
Is this MEF Import always guaranteed to be there? If not then you need to make it optional or the entire LSP Server won't start. Don't ask how I know. :)

Reply:
I believe so, IBrokeredDebuggerServices is guaranteed since this is an export that I control.  For LspWorkspaceRegistrationService, I'm 100% certain that it is as well because other factories (like CodeLensRefreshQueueFactory) don't make it optional.